File: app/src/main/java/com/codingblocks/cbonlineapp/widgets/SheetDialog.java
package com.codingblocks.cbonlineapp.widgets;

import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.util.TypedValue;
import android.view.View;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.StyleRes;

import com.google.android.material.bottomsheet.BottomSheetDialog;

public class SheetDialog extends BottomSheetDialog {

    public SheetDialog(@NonNull Context context, @StyleRes int theme) {
        super(context, getThemeResId(context, theme));
    }

    private static int getThemeResId(Context context, int themeId) {
        if (themeId == 0) {
            TypedValue outValue = new TypedValue();
            if (context.getTheme().resolveAttribute(com.google.android.material.R.attr.bottomSheetDialogTheme, outValue, true)) {
                themeId = outValue.resourceId;
            } else {
                themeId = com.google.android.material.R.style.Theme_Design_Light_BottomSheetDialog;
            }
        }
        return themeId;
    }

    @SuppressWarnings("ConstantConditions")
    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            View v = getWindow().getDecorView();
            int flags = v.getSystemUiVisibility();
            flags &= ~View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
            v.setSystemUiVisibility(flags);
        }
    }
}

File: app/src/main/java/com/codingblocks/cbonlineapp/util/widgets/SheetDialog.kt
package com.codingblocks.cbonlineapp.util.widgets

import android.content.Context
import android.os.Build
import android.os.Bundle
import android.util.TypedValue
import android.view.View
import android.view.WindowManager
import androidx.annotation.StyleRes
import com.google.android.material.bottomsheet.BottomSheetDialog

class SheetDialog(context: Context, @StyleRes theme: Int) : BottomSheetDialog(context, getThemeResId(context, theme)) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window!!.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val v = window!!.decorView
            var flags = v.systemUiVisibility
            flags = flags and View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR.inv()
            v.systemUiVisibility = flags
        }
    }

    companion object {
        private fun getThemeResId(context: Context, themeId: Int): Int {
            var themeId = themeId
            if (themeId == 0) {
                val outValue = TypedValue()
                if (context.theme.resolveAttribute(com.google.android.material.R.attr.bottomSheetDialogTheme, outValue, true)) {
                    themeId = outValue.resourceId
                } else {
                    themeId = com.google.android.material.R.style.Theme_Design_Light_BottomSheetDialog
                }
            }
            return themeId
        }
    }
}
